mitodl/open-discussions
# Extract CourseSearchbox (as Searchbox)

#### Pre-Flight checklist


#### What are the relevant tickets?
Toward #3613 

#### What's this PR do?
This PR extracts CourseSearchbox to a new local package, `ol-search-ui`. It does not add any new functionality to infinite corridor.

Additionally, I set up themes for Styled Components... Good for colors, breakpoints, etc.

#### How should this be manually tested?
1. Visit http://yourname.od.odl.local:8063/learn/ ... make sure the search box still works. In particular, it should be able to be submitted by
    - clicking the magnifier icon
    - tabbing to the magnifier and hitting enter
    - hitting enter on the search input
    - hitting the "Explore" button below search input *In this case, there seems to be an existing bug on rc that the text is not preserved when redirected to search page*
    - **New:** non-empty text should be clearable. *Previously this only worked on the search page*
2. On the search page, the input at top should also work... Check similar things.

    This does fix a small bug where tabbing to the clear icon on RC and hitting enter throws a console error and does not clear the text; now it works.
